presto-main/src/main/java/io/prestosql/cost/PlanNodeLocalCostEstimate.java

```diff
@@ -83,7 +83,8 @@ public PlanCostEstimate add(PlanCostEstimate costEstimate)
     {
         return new PlanCostEstimate(
                 costEstimate.getCpuCost() + this.cpuCost,
-                costEstimate.getMemoryCost() + this.maxMemory,
+                costEstimate.getMaxMemory() + this.maxMemory,
+                costEstimate.getMaxMemoryWhenOutputting() + this.maxMemory,
```

Comment:
this only works for operators that are streaming. For non-streaming operators this should just be `this.maxMemory`
Maybe let's have two `add` specialized methods?

Reply:
Good catch! I moved this method to `CostCalculatorWithEstimatedExchanges` where it's actually used and added explanation why this is OK